### PR TITLE
OBSDOCS-913: add cluster monitoring operator resources api endpoint reference

### PR DESCRIPTION
**Pull request by bburt-rh**
**Note to peer reviewer**: This content is auto-generated from the Operator source code. If you find any style issues, please make a note of them, and I will write up a separate issue to fix the content in the source code.

Version(s): 4.16+

Issue: https://issues.redhat.com/browse/OBSDOCS-913

Link to docs preview: https://71449--ocpdocs-pr.netlify.app/openshift-rosa/latest/observability/monitoring/accessing-third-party-monitoring-apis.html#resources-reference-for-the-cluster-monitoring-operator

QE review:
- [x] QE has approved this change.

Additional information:

This PR adds API endpoint reference information for services and routes for CMO components. This content is autogenerated from the CMO source code.

### CODE REVIEW

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+Expose the `/metrics` endpoint on port 9100. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/openshift-state-metrics
+
+Expose openshift-state-metrics `/metrics` endpoints within the cluster on the following ports:
+
+* Port 8443 provides access to the OpenShift resource metrics. This port is for internal use, and no other usage is guaranteed.
+* Port 9443 provides access to the internal `openshift-state-metrics` metrics. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/prometheus-adapter
```
Comment (juzhao) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:105]: if metrics-server is enabled, there will be no prometheus-adapter service

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+Expose the `prometheus-adapter` web server on port 443. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/prometheus-k8s
```
Comment (juzhao) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:109]: prometheus-k8s service also have 9092 /metrics port, do we need to mention?
```
$ oc -n openshift-monitoring get svc prometheus-k8s
NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
prometheus-k8s   ClusterIP   172.30.178.48   <none>        9091/TCP,9092/TCP   5h47m

$ oc -n openshift-monitoring get svc prometheus-k8s -oyaml
...
  ports:
  - name: web
    port: 9091
    protocol: TCP
    targetPort: web
  - name: metrics
    port: 9092
    protocol: TCP
    targetPort: metrics
...
```

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+=== openshift-monitoring/telemeter-client
+
+Expose the `/metrics` endpoint on port 8443. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/thanos-querier
+
+Expose the Thanos Querier web server within the cluster on the following ports:
+
+* Port 9091 provides access to all the Thanos Querier endpoints. Granting access requires binding a user to the `cluster-monitoring-view` cluster role.
+* Port 9092 provides access to the `/api/v1/query`, `/api/v1/query_range/, `/api/v1/labels`, `/api/v1/label/*/values`, and `/api/v1/series` endpoints restricted to a given project. Granting access requires binding a user to the `view` cluster role in the project.
```
Comment (juzhao) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:141]:
```suggestion
* Port 9092 provides access to the `/api/v1/query`, `/api/v1/query_range/`, `/api/v1/labels`, `/api/v1/label/*/values`, and `/api/v1/series` endpoints restricted to a given project. Granting access requires binding a user to the `view` cluster role in the project.
```

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 2 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:109]: @simonpasquier WDYT?

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 3 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:141]: Fixed

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 1 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:105]: @simonpasquier - WDYT?

Comment (slashpai) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:105]: Ya once metrics server GA epic is completed default would be metrics server. At this time it's not GA yet but we are planning to have it GA in 4.16 🤞

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.17` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.16. And any PR going into `main` must also target the latest version branch (enterprise-4.17).

If the update in your PR does NOT apply to version 4.17 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 1 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc]: Replaced with `metrics-server`

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 2 above)
Comment (simonpasquier) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:109]: I wouldn't document it since it's purely internal.

Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:109]: Added

File: modules/installation-mirror-repository.adoc [hunk 4]
```diff
@@ -225,7 +225,7 @@ mirrored, extract it and pin it to the release:
 +
 [source,terminal]
 ----
-$ oc adm release extract -a ${LOCAL_SECRET_JSON} --icsp-file=<file> \ --command=openshift-install "${LOCAL_REGISTRY}/${LOCAL_REPOSITORY}:${OCP_RELEASE}"
+$ oc adm release extract -a ${LOCAL_SECRET_JSON} --icsp-file=<file> --command=openshift-install "${LOCAL_REGISTRY}/${LOCAL_REPOSITORY}:${OCP_RELEASE}-${ARCHITECTURE}"
```
Comment (juzhao) [inline modules/installation-mirror-repository.adoc:228]: this file is not related to monitoring, but checked with oc client 4.16.0-0.nightly-2024-07-25-145501, there is not `icsp-file` parameter, need the image registry QE to check
```
# oc version
Client Version: 4.16.0-0.nightly-2024-07-25-145501
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3

# oc adm release extract --help | grep -i icsp-file
no result
```

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+* Port 9095 provides access to the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-api-reader` role (for read-only operations) or `monitoritoring-alertmanager-api-writer` role in the `openshift-user-workload-monitoring` project.
+* Port 9092 provides access to the Alertmanager endpoints restricted to a given project. Granting access requires binding a user to the `monitoring-rules-edit` cluster role or `monitoring-edit` cluster role in the project.
+* Port 9097 provides access to the `/metrics` endpoint only. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/alertmanager-main
+
+Expose the Alertmanager web server within the cluster on the following ports:
+
+* Port 9094 provides access to all the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-edit` role in the `openshift-monitoring` project.
```
Comment (juzhao) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:75]: see 9094 port usage
```
$ oc -n openshift-monitoring get svc  alertmanager-main -oyaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    openshift.io/description: |-
      Expose the Alertmanager web server within the cluster on the following ports:
      * Port 9094 provides access to all the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-view` role (for read-only operations) or `monitoring-alertmanager-edit` role in the `openshift-monitoring` project.
      ...
```

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+Expose the `/metrics` endpoint on port 9100. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/openshift-state-metrics
+
+Expose openshift-state-metrics `/metrics` endpoints within the cluster on the following ports:
+
+* Port 8443 provides access to the OpenShift resource metrics. This port is for internal use, and no other usage is guaranteed.
+* Port 9443 provides access to the internal `openshift-state-metrics` metrics. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/metrics-server
```
Comment (juzhao) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:105]: two `openshift-monitoring/metrics-server`

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+Expose the `metrics-server` web server on port 443. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/prometheus-k8s
+
+Expose the Prometheus web server within the cluster on the following ports:
+
+* Port 9091 provides access to all the Prometheus endpoints. Granting access requires binding a user to the `cluster-monitoring-view` cluster role.
+
+* Port 9092 exposes the `/metrics` endpoint. This port is for internal use, and no other usage is guaranteed.
```
Comment (juzhao) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:115]: 9092 usage see:
https://github.com/openshift/cluster-monitoring-operator/blob/release-4.16/Documentation/resources.md?plain=1#L93

File: modules/installation-mirror-repository.adoc (hunk 4 above)
Comment (bburt-rh) [reply modules/installation-mirror-repository.adoc:228]: this was mistakenly included in the PR. Will remove it.

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 5 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:75]: Updated per the source code

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 7 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:115]: Updated per the source

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 6 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc]: removed duplicate

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+* Port 9095 provides access to the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-api-reader` role (for read-only operations) or `monitoritoring-alertmanager-api-writer` role in the `openshift-user-workload-monitoring` project.
+* Port 9092 provides access to the Alertmanager endpoints restricted to a given project. Granting access requires binding a user to the `monitoring-rules-edit` cluster role or `monitoring-edit` cluster role in the project.
+* Port 9097 provides access to the `/metrics` endpoint only. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/alertmanager-main
+
+Expose the Alertmanager web server within the cluster on the following ports:
+
+* * Port 9094 provides access to all the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-view` role (for read-only operations) or the `monitoring-alertmanager-edit` role in the `openshift-monitoring` project.
```
Comment (juzhao) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:75]:
```suggestion
*  Port 9094 provides access to all the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-view` role (for read-only operations) or the `monitoring-alertmanager-edit` role in the `openshift-monitoring` project.
```

Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc]: fixed

**Review by simonpasquier — COMMENTED** [review]
/lgtm

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+=== openshift-monitoring/prometheus-operator-admission-webhook
+
+Expose the admission webhook service which validates `PrometheusRules` and `AlertmanagerConfig` custom resources on port 8443.
+
+=== openshift-user-workload-monitoring/alertmanager-user-workload
+
+Expose the user-defined Alertmanager web server within the cluster on the following ports:
+
+* Port 9095 provides access to the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-api-reader` role (for read-only operations) or `monitoritoring-alertmanager-api-writer` role in the `openshift-user-workload-monitoring` project.
```
Comment (kcarmichael08) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:67]: I think there is a typo here: "`monitoritoring-alertmanager-api-writer`"

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+Expose the `/metrics` endpoint on port 8443. This port is for internal use, and no other usage is guaranteed.
+
+=== openshift-monitoring/thanos-querier
+
+Expose the Thanos Querier web server within the cluster on the following ports:
+
+* Port 9091 provides access to all the Thanos Querier endpoints. Granting access requires binding a user to the `cluster-monitoring-view` cluster role.
+* Port 9092 provides access to the `/api/v1/query`, `/api/v1/query_range/`, `/api/v1/labels`, `/api/v1/label/*/values`, and `/api/v1/series` endpoints restricted to a given project. Granting access requires binding a user to the `view` cluster role in the project.
+* Port 9093 provides access to the `/api/v1/alerts`, and `/api/v1/rules` endpoints restricted to a given project. Granting access requires binding a user to the `monitoring-rules-edit` cluster role or `monitoring-edit` cluster role or `monitoring-rules-view` cluster role in the project.
```
Comment (kcarmichael08) [inline modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:140]:
```suggestion
* Port 9093 provides access to the `/api/v1/alerts`, and `/api/v1/rules` endpoints restricted to a given project. Granting access requires binding a user to the `monitoring-rules-edit` cluster role, `monitoring-edit` cluster role, or `monitoring-rules-view` cluster role in the project.
```

**Review by kcarmichael08 — COMMENTED** [review]
Couple of things that I think should be fixed in the source.

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 9 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:67]: Good catch! I'll go ahead and fix it here, and write up a ticket to fix it in the source, too. Thanks!

File: modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc (hunk 10 above)
Comment (bburt-rh) [reply modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc:140]: I'll go ahead and fix it here, and write up a ticket to fix it in the source, too. Thanks!

**Pull request merged**